src/types/simple.rs

```diff
@@ -148,6 +147,7 @@ impl<T: PrimType> Container<T> {
             Self::Sum(variants) => Container::Sum(map_row(*variants, f)),
             Self::Array(elem, sz) => Container::Array(Box::new(f(*elem)), sz),
             Self::Alias(s) => Container::Alias(s),
+            Self::Opaque(custom) => Container::Opaque(custom),
```

Comment:
This is the tricky one. So if we have `c: Container<ClassicType>` we can make a `Container<SimpleType>` by calling `c.map_vals(&SimpleType::Classic)`. I can't see what else we could do here apart from just return another `Container::Opaque`...